frontend/src/templates/Field/Nric/NricField.tsx

```diff
@@ -28,7 +29,12 @@ export const NricField = ({ schema }: NricFieldProps): JSX.Element => {
         aria-label={`${schema.questionNumber}. ${schema.title}`}
         defaultValue=""
         preventDefaultOnEnter
-        {...register(schema._id, validationRules)}
+        value={value}
+        {...register(schema._id, {
+          ...validationRules,
+          onChange: (event: React.ChangeEvent<HTMLInputElement>) =>
+            setValue(event.target.value.trim()),
+        })}
```

Comment:
So just to confirm the render flow: if a user inputs a leading/trailing space, [validation](https://github.com/opengovsg/FormSG/blob/0f211f486f06c57a3d898dcb136eeb38cd38df7b/frontend/src/utils/fieldValidation.ts#L321-L339) will fail, but `onChange()` will fire and correct the input, validation will run again, and it will pass, and the user will not see any error message flash, correct?

Reply:
Yes to `user will not see any error message flash`.  The flow works as expected for users that is correcting their input, but will fail when user had explicitly added a space. But to be pedantic, it will be called with the data that the user had input as opposed to the output of `setState`.

The reason for why this happens is due to the fact that `react-hooks-form` operates on uncontrolled component, and my initial implementation is turning the component into a controlled component, which breaks the assumption that `react-hooks-form` had.

The correct way to manipulate the form data is to use `setValue` hook provided by the form's context.

Reply:
Thanks for the explanation!